include/circt/Dialect/Calyx/CalyxStructure.td

```diff
@@ -76,9 +75,29 @@ def ComponentOp : CalyxOp<"component", [
       /// Hooks for the input/output type enumeration in FunctionLike.
       unsigned getNumFuncArguments() { return getType().getNumInputs(); }
       unsigned getNumFuncResults() { return getType().getNumResults(); }
+
+      /// This is a helper function that should only be used to get the WiresOp or
+      /// ControlOp of a ComponentOp, which are guaranteed to exist and generally at
+      /// the end of a component's body. In the worst case, this will run in linear
+      /// time with respect to the number of instances within the cell.
+      template <typename Op>
+      Operation* getOpWithType() {
```

Comment:
You can return `Op` instead of `Operation *` to remove some casts in client code of this.

Reply:
I think I should be able to get around this by separating the declarations and definitions.